internal/kube/client.go

```diff
@@ -93,20 +93,19 @@ type ResolvedServicePort struct {
 
 // ResolveServicePorts converts named ports into their true
 // format. TargetPort's that have are named become their integer equivalents
-func ResolveServicePorts(ctx context.Context, k kubernetes.Interface,
-	s *corev1.Service) ([]ResolvedServicePort, bool, error) {
-	store := kevents.GlobalCache.GetStore(&corev1.Endpoints{})
+func ResolveServicePorts(s *corev1.Service) ([]ResolvedServicePort, bool, error) {
+	store := kevents.GlobalCache.Core().V1().Endpoints().Informer().GetStore()
 	if store == nil {
 		return nil, false, fmt.Errorf("endpoints store was empty")
 	}
 
-	obj, exists, err := store.GetByKey(s.Namespace + "/" + s.Name)
+	obj, _, err := store.GetByKey(s.Namespace + "/" + s.Name)
```

Comment:
Does `Get()` with the service being provided here work? I know it's not the same type, but just wondering given we're just feeding it the same thing essentially.

Reply:
The key for all resources is build using this function https://github.com/kubernetes/client-go/blob/0964d4be7536bb9f7fae6bb71fbe763942cd2592/tools/cache/store.go#L99. So its always `namespace + "/" + name`, as Endpoints by definition have the same name as services they belong to this is perfectly fine.